lib/features/thread/presentation/thread_controller.dart

```diff
@@ -74,20 +90,15 @@ class ThreadController extends BaseController {
 
   void _resetPositionCurrentAndLoadMoreState() {
     if (loadMoreState.value == LoadMoreState.LOADING) {
-      final newPosition = positionCurrent - emailList.length;
-      _updatePositionCurrent(newPosition);
+      positionCurrent -= lastGetTotal;
```

Comment:
why not reset to 0?

Reply:
In case of network outage or in some cases not due to location restrictions, users cannot download more data. We need to keep the current location so the user can continue to download more at that location